apis/apiextensions/v1alpha1/xrd_types.go

```diff
@@ -44,58 +64,61 @@ type CompositeResourceDefinitionSpec struct {
 	// +immutable
 	EnforcedCompositionRef *v1alpha1.Reference `json:"enforcedCompositionRef,omitempty"`
 
-	// ClaimNames specifies the names of an optional composite resource claim.
-	// When claim names are specified Crossplane will create a namespaced
-	// 'composite resource claim' CRD that corresponds to the defined composite
-	// resource. This composite resource claim acts as a namespaced proxy for
-	// the composite resource; creating, updating, or deleting the claim will
-	// create, update, or delete a corresponding composite resource. You may add
-	// claim names to an existing CompositeResourceDefinition, but they cannot
-	// be changed once they have been set.
-	ClaimNames *extv1.CustomResourceDefinitionNames `json:"claimNames,omitempty"`
-
-	// CRDSpecTemplate is the base CRD template. The final CRD will have additional
-	// fields to the base template to accommodate Crossplane machinery.
-	CRDSpecTemplate CRDSpecTemplate `json:"crdSpecTemplate,omitempty"`
+	// Versions is the list of all API versions of the defined composite
+	// resource. Version names are used to compute the order in which served
+	// versions are listed in API discovery. If the version string is
+	// "kube-like", it will sort above non "kube-like" version strings, which
+	// are ordered lexicographically. "Kube-like" versions start with a "v",
+	// then are followed by a number (the major version), then optionally the
+	// string "alpha" or "beta" and another number (the minor version). These
+	// are sorted first by GA > beta > alpha (where GA is a version with no
+	// suffix such as beta or alpha), and then by comparing major version, then
+	// minor version. An example sorted list of versions: v10, v2, v1, v11beta2,
+	// v10beta3, v3beta1, v12alpha1, v11alpha2, foo1, foo10. Note that all
+	// versions must have identical schemas; Crossplane does not currently
+	// support conversion between different version schemas.
+	Versions []CompositeResourceDefinitionVersion `json:"versions"`
```

Comment:
If I follow what you're asking - yes. This is how v1 CRDs work too I believe. i.e. If you want one version you provide a single element array with the details of that version.

Reply:
Do you think there is any value in making that a little easier by automatically populating the single element array for you?

Reply:
Possibly, but it would be a tradeoff between succinctness and surprisingness; i.e. we'd need to presume that if you didn't specify any versions you'd want exactly one version (perhaps `v1` as you suggest) that was served, referenceable, and had no schema apart from the fields that were required by all XRs. For our generic `Bucket` XR for example it would be the difference between what we require today:

```yaml
apiVersion: apiextensions.crossplane.io/v1alpha1
kind: CompositeResourceDefinition
metadata:
  name: compositebuckets.common.crossplane.io
spec:
  group: common.crossplane.io
  names:
    categories:
      - crossplane
    kind: CompositeBucket
    plural: compositebuckets
  claimNames:
    categories:
      - crossplane
    kind: Bucket
    plural: buckets
  versions:
  - name: v1alpha1
    served: true
    referenceable: true
```

And what we would require if we inferred a default version: 

```yaml
apiVersion: apiextensions.crossplane.io/v1alpha1
kind: CompositeResourceDefinition
metadata:
  name: compositebuckets.common.crossplane.io
spec:
  group: common.crossplane.io
  names:
    categories:
      - crossplane
    kind: CompositeBucket
    plural: compositebuckets
  claimNames:
    categories:
      - crossplane
    kind: Bucket
    plural: buckets
```

If you ever did want to introduce a `v2` of your API you'd need to make sure to also specify `v1` exactly as Crossplane inferred it should be when there was one version, so while I don't feel super strongly I'm inclined to go with the slightly more explicit version that this PR currently implements.